src/overlays/actors/ovl_En_Wiz_Fire/z_en_wiz_fire.c

```diff
@@ -66,48 +73,48 @@ void EnWizFire_Init(Actor* thisx, PlayState* play) {
     EnWizFire* this = THIS;
 
     Collider_InitAndSetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
-    this->unk_162 = this->actor.params;
+    this->type = EN_WIZ_FIRE_GET_TYPE(&this->actor);
     this->actor.targetMode = 3;
-    this->unk_172 = 10;
-    this->unk_1FC = 255.0f;
+    this->wallCheckTimer = 10;
+    this->alpha = 255.0f;
     this->actor.flags &= ~ACTOR_FLAG_1;
 
     if (!Player_HasMirrorShieldEquipped(play)) {
         this->collider.info.toucher.dmgFlags = 0x20000000;
     }
 
-    switch (this->unk_162) {
-        case 4:
-            this->unk_166 = 1;
+    switch (this->type) {
+        case EN_WIZ_FIRE_TYPE_ICE_MAGIC_PROJECTILE:
+            this->isIceType = true;
             this->collider.info.toucher.damage = 8;
             this->collider.info.toucher.effect = 2;
             this->collider.info.bumper.dmgFlags = (0x1000000 | 0x800 | 0x200 | 0x2);
-            this->unk_162 = 0;
-
-        case 0:
-            if (this->unk_162 == 4) {
-                this->unk_162 = 0;
+            this->type = EN_WIZ_FIRE_TYPE_MAGIC_PROJECTILE;
+            // fallthrough
+        case EN_WIZ_FIRE_TYPE_MAGIC_PROJECTILE:
+            if (this->type == EN_WIZ_FIRE_TYPE_ICE_MAGIC_PROJECTILE) {
+                this->type = EN_WIZ_FIRE_TYPE_MAGIC_PROJECTILE;
                 this->collider.info.toucher.damage = 8;
             }
-
-        case 1:
-        case 3:
-            this->actionFunc = func_80A4984C;
+            // fallthrough
+        case EN_WIZ_FIRE_TYPE_ARCING_MAGIC_PROJECTILE:
+        case EN_WIZ_FIRE_TYPE_REFLECTED_MAGIC_PROJECTILE:
+            this->actionFunc = EnWiz_SetupMoveMagicProjectile;
             break;
 
-        case 2:
-            this->actor.draw = func_80A4BAB4;
-            this->unk_170 = Rand_S16Offset(0, 10000);
-            this->unk_160 = 1;
+        case EN_WIZ_FIRE_TYPE_SMALL_FLAME:
+            this->actor.draw = EnWizFire_DrawSmallFlame;
+            this->smallFlameScroll = Rand_S16Offset(0, 10000);
+            this->action = EN_WIZ_FIRE_ACTION_SMALL_FLAME;
             this->collider.info.toucher.damage = 2;
-            this->actionFunc = func_80A49F38;
+            this->actionFunc = EnWiz_SetupSmallFlame;
     }
```

Comment:
```suggestion
            this->actionFunc = EnWiz_SetupSmallFlame;
            break;
    }
```
every case should have a break, and there should be a default if possible.